components/sync15/src/request.rs

```diff
@@ -349,8 +376,6 @@ where
 
         self.queued.push(b']');
         let batch_id = match &self.batch {
```

Comment:
Extract this into a function that returns a String.

```
#[inline]
fn batch_id(&self) -> String {
  // Return current batch id or "true", if not already in a batch.
  match &self.batch {
    BatchState::NoBatch => "true".into(),
    BatchState::InBatch(ref s) => s.clone(),
  }
}

Reply:
Then, you can get rid of `batch_id.is_some()` for `is_commit` and change the call to `post` to do `Some(self.batch_id())` instead of `batch_id`.